fluentd/lib/filter_parse_json_field/lib/filter_parse_json_field.rb

```diff
@@ -39,17 +40,22 @@ def configure(conf)
     end
 
     def filter_stream(tag, es)
-      return es unless @merge_json_log
-
+      return es unless @merge_json_log or @replace_json_log
       new_es = Fluent::MultiEventStream.new
-
-      es.each { |time, record|
-        record = do_merge_json_log(record)
-
-        new_es.add(time, record)
-      }
-
+      if @merge_json_log
+        es.each { |time, record|
+          record = do_merge_json_log(record)
+          new_es.add(time, record)
+        }
+      end
+      if @replace_json_log
```

Comment:
IDoes this do what you expect:  ` new_es.add(time, record)` puts the record into the stream.  This means if you have both merge and replace enabled you will apply the logic separately to the record and add it to the stream twice.  My expectation is that we would do whatever logic and then at the end put the message into the stream.

Reply:
changed it to `if..elsif..` logic. `@replace_json_log` will not run if `@merge_json_log` is enabled.